build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/CheckApiCompatibility.cs

```diff
@@ -0,0 +1,240 @@
+
+				// Get api definition for previous Api
+				genApiProcess.Start ();
+				content = genApiProcess.StandardOutput.ReadToEnd();
+				genApiProcess.WaitForExit ();
+			}
+
+			ValidateIssues(content);
+		}
+
+		// Validates there is no issue or issues found are acceptable
+		void ValidateIssues (string content)
+		{
+			// Load issues into a dictionary
+			var issuesFound = LoadIssues (content);
+
+			// If no issues were reported we can return here.
+			if (issuesFound == null || issuesFound.Count == 0) {
+				return;
+			}
+
+			// Verify if there is a file with acceptable issues.
+			var acceptableIssuesFile = Path.Combine (ApiCompatibilityPath, $"acceptable-breakages-{ApiLevel}.txt");
+			if (!File.Exists (acceptableIssuesFile)) {
+
+				// We have issues but there is not acceptable breaks file.
+				// Print issues and fail build.
+				Log.LogMessage (MessageImportance.High, content);
+				Log.LogError ($"CheckApiCompatibility found nonacceptable Api breakages for ApiLevel: {ApiLevel}.");
+				return;
+			}
+
+			// Read and Convert the acceptable issues into a dictionary
+			var exceptionRulesFileContent = File.ReadAllText (acceptableIssuesFile);
+			var acceptableIssues = LoadIssues (exceptionRulesFileContent);
+
+			// Now remove all acceptable issues form the dictionary of issues found.
+			if (acceptableIssues != null) {
+				foreach (var item in acceptableIssues) {
+					if (!issuesFound.TryGetValue (item.Key, out HashSet<string> issues)) {
+						// we should always be able to find the assembly that is reporting the issues
+						Log.LogWarning ($"There is an invalid assembly listed on the acceptable breakages file: {item.Key}");
+						continue;
+					}
+
+					foreach (var issue in item.Value) {
+						// we should always be able to remove the issue, if we try to remove an issue that does not exist,
+						// it means the acceptable list is incorrect and should be reported.
+						if (!issues.Remove (issue)) {
+							Log.LogWarning ($"There is an invalid issue listed on the acceptable breakages file: {issue}");
+						}
+					}
+				}
+			}
+
+			var count = 0;
+
+			// Any issue that still exist on issues found means it is a new issue and we should report
+			foreach (var item in issuesFound) {
+				if (item.Value.Count == 0) {
+					continue;
+				}
+
+				Log.LogMessage (MessageImportance.High, item.Key);
+				foreach (var issue in item.Value) {
+					Log.LogMessage (MessageImportance.High, issue);
+					count++;
+				}
+			}
+
+			if (count > 0) {
+				Log.LogMessage (MessageImportance.High, $"Total Issues: {count}");
+				Log.LogError ($"CheckApiCompatibility found nonacceptable Api breakages for ApiLevel: {ApiLevel}.");
+			}
+		}
+
+		// Converts list of issue into a dictinary
+		Dictionary<string, HashSet<string>> LoadIssues (string content)
+		{
+			var issues = new Dictionary <string, HashSet<string>> ();
+			HashSet<string> currentSet = null;
+			var lines = content.Split ('\n');
```

Comment:
This should use `StreamReader.ReadLine()`.

Reply:
Is there a specific line ending issues you foresee? I am not seeing yet the benefit of using StreamReader since I am reading the output of the console execution of Microsoft.DotNet.ApiCompat.exeand I need to read the entire content anyways.
Using the string gives me some other advantages like not worry about dispose the stream once I am done.

I have added the string.Trim()

Reply:
Fixed!